charts/jobs/templates/job.yaml

```diff
@@ -1,11 +1,15 @@
 {{- $v := .Values }}
 {{- $ := . }}
-{{- range $name, $spec := $v.tasks }}
-{{- if and $spec.enabled (eq $spec.type "job") }}
+{{- range $spec := $v.jobs }}
+{{- if and (default true $spec.enabled) (eq $spec.type "job") }}
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: {{ join "-" (list $name  uuidv4) | trunc 63 }}
+  {{- if eq $spec.runPolicy "OnSpecChange" }}
```

Comment:
A spec for an existing job cannot be redeployed. The only way to deploy a job is to ensure that it has a new name.


My proposal for `OnSpecChange` scenario is the following:
To create a job with a name that contains hash from spec. 
In this way we ensure that if there is no change the job won't be run, as it will be deployed with unchanged status, e.g.:
```
job.batch/<job-name> unchanged
```